### PR TITLE
Disallow anonymous access to draft branch #9853

### PR DESCRIPTION
**Pull request by anatol-sialitski**

### CODE REVIEW

File: modules/launcher/launcher-impl/src/main/resources/META-INF/config/system.properties [hunk 1]
```diff
@@ -69,3 +69,8 @@ xp.config.paths=${xp.home}/config
 #
 # xp.init.maxTries = 300
 
+#
+# Global security settings
+#
+# xp.disallow.access.to.draft.branch.for.anonymous = true
```
Comment (rymsha) [inline modules/launcher/launcher-impl/src/main/resources/META-INF/config/system.properties:75]: It should be a portal config. Not a global config.
Config should specify principals who have access

File: modules/portal/portal-api/src/main/java/com/enonic/xp/portal/handler/BasePortalHandler.java [hunk 2]
```diff
@@ -43,6 +43,8 @@ protected WebResponse doHandle( final WebRequest webRequest, final WebResponse w
             ContextAccessor.current().getLocalScope().setAttribute( portalRequest.getRepositoryId() );
             ContextAccessor.current().getLocalScope().setAttribute( portalRequest.getBranch() );
 
+            doHandleBeforeNextInChain( portalRequest );
```
Comment (rymsha) [inline modules/portal/portal-api/src/main/java/com/enonic/xp/portal/handler/BasePortalHandler.java:46]: Avoid new API method. It should be possible to do without extending API

File: modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/handler/portal/SiteHandler.java [hunk 3]
```diff
@@ -32,6 +40,21 @@ protected PortalRequest createPortalRequest( final WebRequest webRequest, final
...
     }
 
+    @Override
+    protected void doHandleBeforeNextInChain( final PortalRequest portalRequest )
+    {
+        if ( !"false".equalsIgnoreCase( System.getProperty( DISALLOW_ACCESS_TO_DRAFT_BRANCH_FOR_ANONYMOUS_CONFIG ) ) &&
+            ContentConstants.BRANCH_DRAFT.equals( portalRequest.getBranch() ) )
+        {
+            final AuthenticationInfo authInfo = ContextAccessor.current().getAuthInfo();
+            if ( !( authInfo.hasRole( RoleKeys.ADMIN ) || authInfo.hasRole( RoleKeys.ADMIN_LOGIN ) ) &&
```
Comment (rymsha) [inline modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/handler/portal/SiteHandler.java:50]: Anonymous should get Authentication Required 
Any other with insufficient rights should get Forbidden

**Review by rymsha — CHANGES_REQUESTED** [review]

**Review by vbradnitski — APPROVED** [review]

File: modules/runtime/src/home/config/com.enonic.xp.portal.cfg [hunk 4]
```diff
@@ -3,3 +3,4 @@
 #media.private.cacheControl = private, max-age=31536000, immutable
 #media.contentSecurityPolicy = default-src 'none'; base-uri 'none'; form-action 'none'
 #media.contentSecurityPolicy.svg = default-src 'none'; base-uri 'none'; form-action 'none'; style-src 'self' 'unsafe-inline'
+#draftBranchAllowedFor = role:system.admin, role:system.admin.login
```
Comment (rymsha) [inline modules/runtime/src/home/config/com.enonic.xp.portal.cfg:6]: delete `role:system.admin` from this list.

**Review by rymsha — APPROVED** [review]

**Pull request merged**